scripts/cat/status.py

```diff
@@ -0,0 +1,639 @@
+        changes.
+        """
+        # checks that we don't add a duplicate group/rank pairing
+        if self.group_history:
+            last_entry = self.group_history[-1]
+            if last_entry["group"] == self.group and last_entry["rank"] == new_rank:
+                return
+
+        self.group_history.append(
+            {"group": self.group, "rank": new_rank, "moons_as": 0}
+        )
+
+    def change_group_nearness(self, group: CatGroup):
+        """
+        Flips the "near" bool of the given group.
+        """
+        for entry in self.standing_history:
+            if entry.get("group") == group:
+                if entry.get("near"):
+                    entry["near"] = False
+                else:
+                    entry["near"] = True
+
+    # RETRIEVE INFO
+    def get_standing_with_group(self, group: CatGroup) -> list[CatStanding]:
+        """
+        Returns the list of standings a cat has for the given group.
+        """
+        standing_list = []
+        for entry in self.standing_history:
+            if entry["group"] == group:
+                standing_list = entry["standing"]
+                break
+
+        return standing_list
+
+    def find_prior_clan_rank(self, clan: CatGroup = None):
+        """
+        Finds the last clan rank held of a current outsider
+        :param clan: pass the name of a clan to only return the cat's prior rank within that clan. Default is None, if
+        None then the last rank within any Clan will be returned.
+        """
+        if clan:
+            past_ranks = [
+                record["rank"]
+                for record in self.group_history
+                if record["group"] == clan
+            ]
+        else:
+            past_ranks = [
+                rank
+                for rank in self.all_ranks.keys()
+                if rank not in [CatRank.LONER, CatRank.KITTYPET, CatRank.ROGUE]
+            ]
+
+        return past_ranks[-1]
+
+    def in_player_clan(self) -> bool:
+        """
+        Returns True if the cat is currently part of the player clan.
+        """
+        return True if self.group == CatGroup.PLAYER_CLAN else False
+
+    def is_outsider(self) -> bool:
+        """
+        Returns True if the cat isn't part of a clan.
+        """
+
+        return True if self.social != CatSocial.CLANCAT else False
+
+    def is_clancat(self) -> bool:
+        """
+        Returns True if the cat is currently a clancat in any clan.
+        """
+        return True if self.social == CatSocial.CLANCAT else False
+
+    def is_former_clancat(self) -> bool:
+        """
+        Returns True if the cat has been part of any clan in the past, but is not currently a clancat.
+        """
+
+        return (
+            True
+            if CatSocial.CLANCAT in self.all_socials
+            and self.social != CatSocial.CLANCAT
+            else False
+        )
```

Comment:
```suggestion
        return (CatSocial.CLANCAT in self.all_socials
            and self.social != CatSocial.CLANCAT)
```
May not need the parentheses, might accidentally make that into a tuple 😬 please check